File: files/2026-04-12/CSEC 12042026.py
import pandas as pd

path = "2026-04-12/Challenge 112.xlsx"
input = pd.read_excel(path, usecols="B", skiprows=1, nrows=5).rename(columns=lambda c: c.strip() if isinstance(c, str) else c)
test = pd.read_excel(path, usecols="D:E", skiprows=1, nrows=5).rename(columns=lambda c: c.strip() if isinstance(c, str) else c)

result = (
	input
	.assign(rn=lambda d: range(1, len(d) + 1))
	.assign(Name=lambda d: d["Name"].astype(str).str.split(" "))
	.explode("Name")
	.assign(
		label=lambda d: d["Name"].str.fullmatch(r"[A-Z]+", na=False).map(
			{True: "First Name", False: "Other Names"}
		)
	)
	.groupby(["rn", "label"], as_index=False)["Name"]
	.agg(" ".join)
	.pivot(index="rn", columns="label", values="Name")
	.reset_index(drop=True)
	.reindex(columns=["First Name", "Other Names"])
)
result.columns.name = None

print(result.equals(test))
# True

File: files/2026-04-12/CSEC 12042026.R
library(tidyverse)
library(readxl)

path <- "2026-04-12/Challenge 112.xlsx"
input <- read_excel(path, range = "B2:B6")
test <- read_excel(path, range = "D2:E6")

result <- input %>%
  mutate(rn = row_number()) %>%
  separate_rows(Name, sep = " ") %>%
  mutate(label = if_else(str_detect(Name, "^[A-Z]+$"), "First Name", "Other Names")) %>%
  summarise(Name = str_c(Name, collapse = " "), .by = c(rn, label)) %>%
  pivot_wider(names_from = label, values_from = Name) %>%
  select(-rn)

all.equal(result, test)
## [1] TRUE
